### PR TITLE
Use cliopts.Http for OAuth 2.0 middleware

### PR DESCRIPTION
**Pull request by aherrmann-da**
- Use `com.daml.cliopts.Http` for OAuth 2.0 middleware host and port command-line flags.
  - This is consistent with the flags that the trigger service offers.
  - This allows configuring the list host to something else than "localhost", e.g. "0.0.0.0".
  - This adds a `--port-file` flag as well.
- Add test-case for the `--port-file` option.
- Update the docs.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala [hunk 1]
```diff
@@ -65,10 +70,12 @@ object Config {
     new scopt.OptionParser[Config]("oauth-middleware") {
       head("OAuth2 Middleware")
 
-      opt[Int]("port")
-        .action((x, c) => c.copy(port = Port(x)))
-        .required()
-        .text("Port to listen on")
+      cliopts.Http.serverParse(this, serviceName = "Trigger")(
```
Comment (cocreature) [inline triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala:73]:
```suggestion
      cliopts.Http.serverParse(this, serviceName = "OAuth2 Middleware")(
```

File: triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala [hunk 2]
```diff
@@ -36,13 +38,16 @@ case class Config(
 )
 
 object Config {
+  val DefaultHttpPort: Int = 3000
```
Comment (cocreature) [inline triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala:41]: any particular reason to choose 3000? I don’t really understand any of our default port numbers so no issue with that just curious.

**Review by cocreature — APPROVED** [review]
lgtm, thanks!

File: triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala (hunk 2 above)
Comment (aherrmann-da) [reply triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Config.scala:41]: No, no particular reason. This just happened to be the port number used for the oauth middleware in the docs examples.

**Pull request merged**